htdocs/main.inc.php

```diff
@@ -1694,6 +1694,11 @@ function top_httphead($contenttype = 'text/html', $forcenocache = 0)
 			$hookmanager = new HookManager($db);
 		}
 		$hookmanager->initHooks(array("main"));
+		$parameters = array();  // @phan-suppress-current-line PhanPluginRedundantAssignment
+		$reshook = $hookmanager->executeHooks('beforeBodyOpen', $parameters); // Note that $action and $object may have been modified by some hooks
```

Comment:
If you want to intercept or execute hook here with the same condtion (so only in some cases of setup), you have already a hook 2 lines after.
I think what you want to do is having a hook global for the top_httphead() that is always called (before outputing the http header or after outputing the http header). 
If i am right, you should set the hoop at begin or at end of top_htthead() and not in the middle inside a line executed if a condition is matched.